### PR TITLE
Codehealth remove base value remaining

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This PR aims to remove the remaining deprecated uses of `base::Value` across the codebase. This is mailing the use of `base::Value` as a `base::Value::Dict` instance, through deprecated functions (e.g. `FindListKey`, `FindStringKey`, etc).

Resolves https://github.com/brave/brave-browser/issues/30014

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/p3a/message_manager_unittest.cc [hunk 1]
```diff
@@ -214,18 +214,20 @@ class P3AMessageManagerTest : public testing::Test,
   void StoreJsonMetricInMap(const network::ResourceRequest& request,
                             bool is_p2a) {
     base::StringPiece body = ExtractBodyFromRequest(request);
-    base::Value parsed_log = *base::JSONReader::Read(body);
-    std::string metric_name = *parsed_log.FindStringKey("metric_name");
-    int metric_value = *parsed_log.FindIntKey("metric_value");
+    base::Value::Dict parsed_log = base::test::ParseJsonDict(body);
+    auto* metric_name = parsed_log.FindString("metric_name");
```
Comment (iefremov) [inline components/p3a/message_manager_unittest.cc:218]: nit: i think explicit types are better than auto here

**Review by iefremov — APPROVED** [review]
components/p3a lgtm

File: components/p3a/message_manager_unittest.cc (hunk 1 above)
Comment (cdesouza-chromium) [reply components/p3a/message_manager_unittest.cc]: Fixed

**Review by goodov — APPROVED** [review]

**Review by aseren — APPROVED** [review]
Brave ads change LGTM

File: browser/brave_wallet/keyring_service_unittest.cc [hunk 2]
```diff
@@ -3901,14 +3900,13 @@ class KeyringServiceAccountDiscoveryUnitTest : public KeyringServiceUnitTest {
...
-    absl::optional<base::Value> request_value =
-        base::JSONReader::Read(request_string);
-    if (*request_value->GetDict().FindString("method") ==
-        "eth_getTransactionCount") {
-      base::Value* params = request_value->FindListKey("params");
-      EXPECT_TRUE(params);
-      std::string* address = params->GetList()[0].GetIfString();
-      EXPECT_TRUE(address);
+    base::Value::Dict dict = base::test::ParseJsonDict(request_string);
+    ASSERT_TRUE(dict.FindString("method"));
```
Comment (yrliou) [inline browser/brave_wallet/keyring_service_unittest.cc:3904]: Save the result and use it below.

**Review by yrliou — APPROVED** [review]

**Review by darkdh — APPROVED** [review]

File: browser/brave_wallet/keyring_service_unittest.cc (hunk 2 above)
Comment (cdesouza-chromium) [reply browser/brave_wallet/keyring_service_unittest.cc]: Done.

**Pull request merged**